File: prototypeLeger/app/src/main/java/com/example/scrabbleprototype/model/Notification.kt
package com.example.scrabbleprototype.model

import kotlinx.serialization.Serializable
import java.text.SimpleDateFormat
import java.util.*

@Serializable
class Notification(var type: Int, var sender: String, var description: String) {
    var title: String = ""
    var date: String = getDateStamp()
    var time: String = getTimeStamp()

    init {
        when(type) {
            0 -> {
                title = "Invitation d'ami"
            }
            1 -> {
                title = "Nouveau message"
            }
        }
    }
    fun getDateStamp(): String {
        val dateFormat = SimpleDateFormat("dd/M/yy", Locale.getDefault())
        return dateFormat.format(Date())
    }

    fun getTimeStamp(): String {
        val time = Date().time
        val timestampFormat = SimpleDateFormat("HH:mm:ss", Locale.getDefault())
        return timestampFormat.format(time)
    }
}

File: common/notification.ts

export enum NotifType {
    Friend,
    Game,
    Message,
}

export class Notification {
    type: NotifType;
    sender: string;
    description: string;
    title: string;
    date: string;
    time: string;
    _id?: string;

    constructor(type: NotifType, sender: string, description: string) {
        this.type = type;
        this.sender = sender;
        this.description = description;
        switch(type) {
            case NotifType.Friend: this.title = "Invitation d'ami";
            case NotifType.Game: this.title = "Invitation à une partie";
            case NotifType.Message: this.title = "Nouveau message";
        }
        this.date = new Date().getFullYear().toString(),
        (new Date().getMonth() + 1).toString().padStart(2, '0'),
        (new Date().getDate().toString().padStart(2, '0'))

        this.time = new Date().getHours().toString().padStart(2, '0') +
        ':' +
        new Date().getMinutes().toString().padStart(2, '0') +
        ':' +
        new Date().getSeconds().toString().padStart(2, '0')
    }
}